tests/common.rs

```diff
@@ -124,27 +127,67 @@ pub fn setup_server() -> u16 {
             let (socket_addr, server) = start_server();
             tx.send(socket_addr.port())
                 .expect("To be able to send port");
-            server.await
+            server.await.expect("To start the server")
         });
     });
 
     rx.recv().expect("To get the bound port.")
 }
 
 /// Configures and starts the server
-fn start_server() -> (SocketAddr, impl Future<Output = ()> + 'static) {
+fn start_server() -> (SocketAddr, impl Future<Output = hyper::Result<()>> + 'static) {
     let socket_addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 0);
+
     const ASSETS_DIR: &str = "tests/test_html";
```

Comment:
Let's hoist this to the top of the file, and then use it directly from inside `handle_file_request` with:
```rust
let asset_file = Path::new(ASSETS_DIR).join(uri_path);
```

Reply:
Sounds good, that makes the service creation easier to.